include/kllvm/ast/AST.h

```diff
@@ -110,13 +135,39 @@ class KOREObjectSymbol : public KORESymbol {
     return arguments;
   }
   const KOREObjectSort *getSort() const { return sort; }
+  uint32_t getTag() const { assert(firstTag == lastTag); return firstTag; }
+  uint16_t getLayout() const { return layout; }
 
   virtual void print(std::ostream &Out, unsigned indent = 0) const override;
 
+  bool operator==(KOREObjectSymbol other) const;
+
+  std::string layoutString() const;
+  uint8_t length() const;
+
+  bool isConcrete() const;
+  bool isPolymorphic() const;
+  bool isBuiltin() const;
+
+  void instantiateSymbol(KOREObjectSymbolDeclaration *decl);
+
+  friend HashSymbol;
+
+  friend KOREDefinition;
+
 private:
   KOREObjectSymbol(const std::string &Name) : name(Name), sort(nullptr) { }
 };
 
+struct HashSymbol {
+  size_t operator()(const kllvm::KOREObjectSymbol &s) const noexcept {
+    size_t hash = 0;
+    boost::hash_combine(hash, s.name);
+    boost::hash_combine(hash, s.arguments);
+    return hash;
+  }
+};
```

Comment:
I'm curious why this is a struct. It doesn't have any members, and is being used as a function object, and it's not static, so you must instantiate a stateless object to use it. Why not just use a namespace?

Reply:
This is typically used when one wants to use such a struct as a template argument, e.g. for an unordered map. The alternative is to use the C++ lambdas which looks kind of awkward.